Add a section on using nextest (#2811)
rust-lang/miri: README.md

  reduced feature set. We might ship Miri with a nightly even when some features
  on these targets regress.

### Running tests in parallel

Though it implements Rust threading, Miri itself is a single-threaded interpreter.
This means that when running `cargo miri test`, you will probably see a dramatic
increase in the amount of time it takes to run your whole test suite due to the
inherent interpreter slowdown and a loss of parallelism.

You can get your test suite's parallelism back by running `cargo miri nextest run`

> (note that you will need `cargo-nextest` installed). This works because `cargo-nextest`

Reviewer: Please add a link to the cargo-nextest website here.

  reduced feature set. We might ship Miri with a nightly even when some features
  on these targets regress.

### Running tests in parallel

Though it implements Rust threading, Miri itself is a single-threaded interpreter.
This means that when running `cargo miri test`, you will probably see a dramatic
increase in the amount of time it takes to run your whole test suite due to the
inherent interpreter slowdown and a loss of parallelism.

You can get your test suite's parallelism back by running `cargo miri nextest run`
(note that you will need [`cargo-nextest`](https://nexte.st) installed).
This works because `cargo-nextest` collects a list of all tests then launches a

> separate `cargo miri run` for each test.

Reviewer: According to https://nexte.st/book/miri.html, by default this will not actually result in any parallelism. So probably we should link to https://nexte.st/book/miri.html, and mention `-jN`, and also warn that this can eat huge amounts of RAM (which is the reason for that default AFAIK)?

Author: I'm attempting to punt to the nextest documentation for a justification. I think Rain is just citing the behavior of `cargo miri test` which seems a bit dubious as justifications for defaults go. Memory makes more sense.

  reduced feature set. We might ship Miri with a nightly even when some features
  on these targets regress.

### Running tests in parallel

Though it implements Rust threading, Miri itself is a single-threaded interpreter.
This means that when running `cargo miri test`, you will probably see a dramatic
increase in the amount of time it takes to run your whole test suite due to the
inherent interpreter slowdown and a loss of parallelism.

You can get your test suite's parallelism back by running `cargo miri nextest run -jN`
(note that you will need [`cargo-nextest`](https://nexte.st) installed).
This works because `cargo-nextest` collects a list of all tests then launches a

> separate `cargo miri run` for each test. You will need to specify a `-j` or `--test-threads`,
> by default `cargo miri nextest run` runs one test at a time. You can find the `cargo-nextest`
> documentation for its Miri support here: https://nexte.st/book/miri.html

Reviewer: ```suggestion
separate `cargo miri run` for each test. You will need to specify a `-j` or `--test-threads`;
by default `cargo miri nextest run` runs one test at a time. For more details, see the
[`cargo-nextest` Miri documentation](https://nexte.st/book/miri.html).
```
